src/coreclr/src/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -11,117 +11,174 @@
 ===========================================================*/
 
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Math
     {
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern double Abs(double value);
+        [SuppressGCTransitionAttribute]
+        [DllImport(RuntimeHelpers.QCall)]
+        private static extern double AbsInternal(double value);
+        public static double Abs(double value) => AbsInternal(value);
 
         [Intrinsic]
```

Comment:
The intrinsic attribute should stay on the public entrypoint.

Reply:
Thanks for the catch. Although with:

```c#
 #pragma warning disable CA1401 // P/Invokes should not be visible 
```
it compiled (and normally works with regular P/Invoke), but checked builds were raising this assert for QCall: https://github.com/dotnet/runtime/blob/867ad8d20402e3bee580e0af6577f87912103c78/src/coreclr/src/vm/ecall.cpp#L724
This was the main reason of defining *Internal variants.

Reply:
The comment in front the QCall assert talks about Code-Access-Security that is not relevant in .NET Core. But it think it may be still a good idea to avoid exporting DllImports directly.